tests/Integration/Behaviour/Features/Scenario/Discount/BO/add_free_gift_discount.feature

```diff
@@ -0,0 +1,51 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s discount --tags add-free-gift-discount
+@restore-all-tables-before-feature
+@restore-languages-after-feature
+@add-free-gift-discount
+Feature: Add discount
+  PrestaShop allows BO users to create discounts
+  As a BO user
+  I must be able to create discounts
+
+  Background:
+    Given shop "shop1" with name "test_shop" exists
+    Given there is a currency named "usd" with iso code "USD" and exchange rate of 0.92
+    Given there is a currency named "chf" with iso code "CHF" and exchange rate of 1.25
+    Given currency "usd" is the default one
+    And language with iso code "en" is the default one
+    And language "french" with locale "fr-FR" exists
+
+  Scenario: Create a simple discount with free gift
+    When I create a free gift discount "basic_free_gift_discount"
+    Then discount "basic_free_gift_discount" should have the following properties:
+      | active        | false |
+
+  Scenario: Create a complete discount with free gift
```

Comment:
You must use gift product ID values and assert they were correctly saved after Same thing with a combination The handler should assert that the provided product ID and combination ID are valid and exist (so use the `ProductRepository::assertProductExists` and `CombinationRepository::assertCombinationExists` methods in the `DiscountValidator`)

By the way this validator has not been created yet so it's a good occasion to introduce it, for now we have nearly no validations on the discount creations